tutorials/intro/plot_object_annotations.py

```diff
@@ -1,245 +1,238 @@
+# function :func:`mne.events_from_annotations` on the :class:`~mne.io.Raw` file
+# containing the annotations. This function will assign an integer Event ID to
+# each unique element of ``raw.annotations.description``, and will return the
+# mapping of descriptions to integer Event IDs along with the derived Event
+# array. By default, one event will be created at the onset of each annotation;
+# this can be modified via the ``chunk_duration`` parameter of
+# :func:`~mne.events_from_annotations` to create equally spaced events within
+# each annotation span. See the function documentation for further details.
+
+events_from_annot, event_dict = mne.events_from_annotations(eeglab_raw)
+print(event_dict)
+print(events_from_annot[:5])
+
+###############################################################################
+# If you want to control which integers are mapped to each unique description
+# value, you can pass a :class:`dict` specifying the mapping as the
+# ``event_id`` parameter of :func:`~mne.events_from_annotations`; this
+# :class:`dict` will be returned unmodified as the ``event_dict``.
 #
+# .. TODO add this when the other tutorial is nailed down:
+#    Note that this ``event_dict`` can be used when creating
+#    :class:`~mne.Epochs` from :class:`~mne.io.Raw` objects, as demonstrated
+#    in :doc:`epoching_tutorial_whatever_its_name_is`.
+
+custom_mapping = {'rt': 77, 'square': 42}
+(events_from_annot,
+ event_dict) = mne.events_from_annotations(eeglab_raw, event_id=custom_mapping)
+print(event_dict)
+print(events_from_annot[:5])
+
+###############################################################################
+# To make the opposite conversion (from Events array to
+# :class:`~mne.Annotations` object), you can create a mapping from integer
+# Event ID to string descriptions, and use the :class:`~mne.Annotations`
+# constructor to create the :class:`~mne.Annotations` object, and use the
+# :meth:`~mne.io.Raw.set_annotations` method to add the annotations to the
+# :class:`~mne.io.Raw` object. Because the :ref:`sample data <sample-dataset>`
+# was recorded on a Neuromag system (where sample numbering starts when the
+# acquisition system is initiated, not when the *recording* is initiated), we
+# also need to pass in the ``orig_time`` parameter so that the onsets are
+# properly aligned relative to the start of recording:
+
+mapping = {1: 'auditory/left', 2: 'auditory/right', 3: 'visual/left',
+           4: 'visual/right', 5: 'smiley', 32: 'buttonpress'}
+onsets = events[:, 0] / raw.info['sfreq']
+durations = np.zeros_like(onsets)  # assumes instantaneous events
+descriptions = [mapping[event_id] for event_id in events[:, 2]]
+annot_from_events = mne.Annotations(onset=onsets, duration=durations,
+                                    description=descriptions,
+                                    orig_time=raw.info['meas_date'])
+raw.set_annotations(annot_from_events)
```

Comment:
I would stop at `annot_from_events = mne...` there is no need to add it to the raw object. And I would not pass `orig_time` or if we want to make it explicit I would do `orig_time=None`.

Reply:
don't you think it's better practice to attach annotations to the raw object so it stays in sync?

Reply:
@massich `orig_time=None` will give the wrong result here, unless you first subtract `raw.first_samp` from the sample numbers in the events array.  Attaching the annotation to the `raw` provides a useful check (since it will issue a warning if any events are outside the time range of `raw`). I think it's also nice to add a `raw.plot()` here, to show the annotations showing up automatically.